src/util/codec/mysql/json.rs

```diff
@@ -563,4 +673,46 @@ mod test {
             assert!(resp.is_err());
         }
     }
+
+    #[test]
+    fn test_cmp_json_between_same_type() {
+        let test_cases = vec![
+            ("false", "true"),
+            ("3", "5"),
+            ("3.0", "4.9"),
+            (r#""hello""#, r#""hello, world""#),
+            (r#"["a", "b"]"#, r#"["a", "c"]"#),
+            (r#"{"a": "b"}"#, r#"{"a": "c"}"#),
+        ];
+        for (left_str, right_str) in test_cases {
+            let left: Json = left_str.parse().unwrap();
+            let right: Json = right_str.parse().unwrap();
+            assert!(left < right);
+            assert_eq!(left, left);
+        }
+        assert_eq!(Json::None, Json::None);
+    }
+
+    #[test]
+    fn test_cmp_json_between_diff_type() {
+        let test_cases = vec![
+            ("1.5", "2"),
+            ("1.5", "false"),
+            ("true", "1.5"),
+            ("true", "2"),
```

Comment:
it's not compatible with MySQL 5.7, although we don't have better implement. But we should comment it here.